Porting to ROS2 Humble

juchajam opened the pull request:
Hi.
I ported this package to ROS2 Humble.
Please review my code and create humble branch.

juchajam committed:
update for ros2

juchajam committed:
fix error and update readme

koide3 commented:
Great PR! Thanks a lot for your contribution.
I'm fully occupied with works in next a few weeks, but I'll try to find a time to review it and create a new branch. Also, I'll add a CI for this later.

alisys-robotics commented:
We are currently using this fork and it works fine. Only required minor changes on hdl_localization.rviz

DCYzhao commented:
Compared with ROS1, this version has a CPU usage three times higher under the same data package and map.

juchajam committed:
perf: optimize CPU utilization and adjust global map publishing

- Switch to `component_container_mt` to resolve thread contention and significantly reduce CPU usage.
- Update `globalmap` publisher to use latched publishing (transient local QoS).
- Disable `use_intra_process_comms` as ROS 2 does not support it with transient_local QoS.
